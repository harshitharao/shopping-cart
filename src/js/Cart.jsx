import React, { Component } from 'react';

class Cart extends Component {
  static defaultProps = { selectedProducts: [] }
  render() {
    const { selectedProducts } = this.props;
    return (
      <div>
        <ul>
          {selectedProducts.map((product, index) => {
            return <li key={index}>{product.name}  {product.price}</li>
          })}
        </ul>
        <p>Total: {selectedProducts.reduce((acc, product) => { return acc + product.price}, 0).toFixed(2)}</p>
      </div>
    );
  }
}

export default Cart;
