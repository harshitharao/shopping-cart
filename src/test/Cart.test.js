import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Cart from '../js/Cart';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Cart />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('cart', () => {
  it('should render empty cart', () => {
    const wrapper = shallow(<Cart selectedProducts={[]}/>);
    expect(wrapper.find('li')).toHaveLength(0);
    expect(wrapper.find('p').text()).toContain('Total: 0');
  });

  it('should render cart with products selected', () => {
    const selectedProducts = [
      { name: 'Dove Soap', price: 39.99},
      { name: 'Dove Soap', price: 39.99},
      { name: 'Dove Soap', price: 39.99},
      { name: 'Dove Soap', price: 39.99},
      { name: 'Dove Soap', price: 39.99}
    ];
    const wrapper = shallow(<Cart selectedProducts={selectedProducts}/>);
    expect(wrapper.find('li')).toHaveLength(5);
    expect(wrapper.find('p').text()).toEqual('Total: 199.95');
  });

  it('should match the cart snapshot when cart is empty', () => {
    const cartTree = renderer.create(<Cart selectedProducts={[]}/>);
    expect(cartTree.toJSON()).toMatchSnapshot();
  });

  it('should match the cart snapshot when cart is has products', () => {
    const cartTree = renderer.create(<Cart selectedProducts={[{ name: 'Dove Soap', price: 39.99}]}/>);
    expect(cartTree.toJSON()).toMatchSnapshot();
  });
});
