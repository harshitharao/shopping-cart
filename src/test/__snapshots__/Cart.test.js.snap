// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart should match the cart snapshot when cart is empty 1`] = `
<div>
  <ul />
  <p>
    Total: 
    0.00
  </p>
</div>
`;

exports[`cart should match the cart snapshot when cart is has products 1`] = `
<div>
  <ul>
    <li>
      Dove Soap
        
      39.99
    </li>
  </ul>
  <p>
    Total: 
    39.99
  </p>
</div>
`;
